### PR TITLE
Update hammer_cli packaging to include prerelease

### PR DESCRIPTION
**Pull request by ehelms**
For plugin updates, please indicate which repos this should be built into:

* [x] Nightly
* [ ] 1.19
* [ ] 1.18
* [ ] 1.17

See Foreman's [plugin maintainer documentation](https://projects.theforeman.org/projects/foreman/wiki/How_to_Create_a_Plugin#Release-strategies) for more information.

---

### CODE REVIEW

File: packages/foreman/rubygem-hammer_cli/rubygem-hammer_cli.spec [hunk 1]
```diff
@@ -57,12 +60,31 @@ Obsoletes: rubygem-%{gem_name}-doc < 0.3.0-2
...
 
 %prep
-%setup -n %{pkg_name}-%{version} -q -c -T
+%{?scl:scl enable %{scl} - << \EOF}
+gem unpack %{SOURCE0}
+%{?scl:EOF}
+
+%setup -q -D -T -n  %{gem_name}-%{version}%{?prerelease}
+
 %{?scl:scl enable %{scl} - <<EOF}
```
Comment (ekohl) [inline packages/foreman/rubygem-hammer_cli/rubygem-hammer_cli.spec]: This shouldn't be here according to the `gem2rpm/scl.spec.erb` template. Is it really needed? If so, do we want to modify our template to allow prereleases for all gems?

**Comment by ekohl** [comment]
Can we give hammer_cli_katello the same treatment so they're really the same? That would save us a lot of trouble.

**Comment by ehelms** [comment]
@ekohl To do so for hammer_cli_katello requires changing how h-c-k handles it source repository. It does not presently include any notion of branch in its version.

**Comment by ekohl** [comment]
Couldn't it use `.pre.master` to reflect it's from the master branch?

**Comment by ehelms** [comment]
Interestingly, the 'pre' part is automatically added by Ruby when using 'gem build' if it finds non-numeric characters at the end of the version. As for adding the branch into the version, thats up to @akofink or other Katello CLI maintainers to decide. If they go that route I am happy to update the spec.

**Comment by akofink** [comment]
Yes, my main goal at this point is to try to package hammer-cli-katello exactly the same as hammer-cli-foreman. I can't see why there should be any differences. Please update the h-c-k spec with this PR. :)

**Comment by ehelms** [comment]
[test rpm]

**Pull request merged**